File: bddtests/src/main/kotlin/mocking/vision/models/ServiceDefinition.kt
package mocking.vision.models

import javax.xml.bind.annotation.XmlElement

data class ServiceDefinition(
        @XmlElement(namespace = "urn:vision")
        var name: String,
        @XmlElement(namespace = "urn:vision")
        var version: String
) {
    constructor() : this("", "")
}

File: xamarinintegrationtests/NHSOnline.HttpMocks/Vision/Models/ServiceDefinition.cs
using System.ComponentModel.DataAnnotations;
using System.Xml.Serialization;

namespace NHSOnline.HttpMocks.Vision.Models
{
    [XmlRoot(ElementName = "serviceDefinition", Namespace = "urn:vision")]
    public sealed class ServiceDefinition
    {
        [XmlElement(ElementName = "name", Namespace = "urn:vision")]
        [Required]
        public string? Name { get; set; }

        [XmlElement(ElementName = "version", Namespace = "urn:vision")]
        [Required]
        public string? Version { get; set; }
    }
}